frontend/src/lib/components/LemonTable/LemonTable.tsx

```diff
@@ -19,7 +19,7 @@ function determineColumnKey(column: LemonTableColumn<any, any>, obligationReason
 function determineColumnKey(column: LemonTableColumn<any, any>, obligationReason?: undefined): string | null
 function determineColumnKey(column: LemonTableColumn<any, any>, obligationReason?: string): string | null {
     const columnKey = column.key || column.dataIndex
-    if (obligationReason && !columnKey) {
+    if (obligationReason && columnKey == null) {
```

Comment:
`dataIndex={0}` didn't work, this is the fix 🙈

Reply:
`== null` or `=== null`?

Reply:
This is intentional, as `foo == null` === `foo === null || foo === undefined`. Probably should have added a comment to clarify this

Reply:
I think it's better to be explicit with these, otherwise it looks like a bug, and a future refactor will not know you intended to catch both values.

Reply:
Yeah, I agree: #11303